apps/browser/src/vault/fido2/content/messaging/messenger.ts

```diff
@@ -127,4 +103,83 @@ export class Messenger {
       localPort.close();
     }
   }
+
+  private createMessageEventListener() {
+    return async (event: MessageEvent<MessageWithMetadata>) => {
+      const windowOrigin = window.location.origin;
+      if (event.origin !== windowOrigin) {
+        return;
+      }
+
+      if (this.handler === undefined) {
+        return;
+      }
```

Comment:
```suggestion
      if (event.origin !== windowOrigin || !this.handler) {
        return;
      }
```
Nit, changes not required - You could consider simplifying these return statements into a single conditional.